docs/lang/logic.md

```diff
@@ -0,0 +1,458 @@
+# Logic
+
+[[Back to all operators]](./standard-operators.md)
+
+In this section, you find the operators that, together with [Sets](./sets.md)
+form the foundation of TLA+. It is a bit strange that we call this section
+"Logic", as the whole language of TLA+ is a logic. However, the operators
+of this section can be often seen in first-order logic, as opposite to the
+propositional logic, see [Booleans](./booleans.md).
+
+Note that the special form `\E y \in S: x' = y` is often used to express
+non-determinism in TLA+. See [Control Flow and Non-determinism]. In this
+section, we only consider the deterministic version.
+
+----------------------------------------------------------------------------
+
+<a name="forallBounded"></a>
+### Bounded universal quantifier
+
+**Notation:** `\A x \in S: P`
+
+**LaTeX notation:** ![forall-bounded](./img/forall-bounded.png)
+
+**Arguments:** At least three arguments: a variable name, a set, and an
+expression. As usual in TLA+, if the second argument is not a set, the result is
+undefined. You can also use multiple variables and tuples, see **Advanced
+syntax**. 
+
+**Effect:** This operator evaluates to a Boolean value. We explain
+semantics only for a single variable:
+
+ - `\A x \in S: P` evaluates to `TRUE`, if for every element `e` of `S`, the
+   expression `P` evaluates to `TRUE` against the binding `[x |-> e]`.
+ - Conversely, `\A x \in S: P` evaluates to `FALSE`, if there exists an element
+   `e` of `S` that makes the expression `P` evaluate to `FALSE` against the
+   binding `[x |-> e]`.
+
+ _Importantly, when `S = {}`, the expression `\A x \ in S: P` evaluates to
+ `TRUE`, independently of what is written in `P`. Likewise, when `{x \in S: P}
+ = {}`, the expression `\A x \ in S: P` evaluates to `TRUE`._
+   
+**Determinism:** Deterministic.
+
+**Errors:** Pure TLA+ does not restrict the operator arguments.  TLC flags a
+model checking error, if `S` is infinite.  Apalache produces a static type
+error, if the type of elements of `S` is not compatible in the context of `P`
+when an element of `S` is bound to `x`.
+
+**Advanced syntax:** Instead of a single variable `x`, you can use the tuple
+syntax to unpack variables from a Cartesian product, see [Tuples](./tuples.md).
+For instance, one can write `\A <<x, y>> \in S: P`. In this case, for every
+element `e` of `S`, the variable `x` is bound to `e[1]` and `y` is bound to
+`e[2]`. The predicate `P` is evaluated against this binding.
+
+Moreover, instead of introducing one variable, one can quantify over several
+sets. For instance, you can write: `\A x \in S, y \in T: P`. This form is
+simply syntax sugar for the form with nested quantifiers: `\A x \in S: \A y
+\in T: P`.
+
```

Comment:
\+ Similarly, `\A x,y \in S: P` is syntax sugar for `\A x \in S : \A y \in S: P`.